### PR TITLE
Better xml docs

### PR DESCRIPTION
**Pull request by GlebChili**
Partially solves #68. Only `API Documentation` section is covered. `Articles` will be added in next PRs. 

Rendered version of the documentation is available at  https://docs.gmodnet.xyz/

### CODE REVIEW

File: gm_dotnet_managed/Tests/GetUserTypeOnTypeMismatch.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
+﻿using GmodNET.API;
```
Comment (SupinePandora43) [inline gm_dotnet_managed/Tests/GetUserTypeOnTypeMismatch.cs:1]: Doesn't seem related to documentation

Comment (GlebChili) [reply gm_dotnet_managed/Tests/GetUserTypeOnTypeMismatch.cs:1]: This test was added to ensure that documented behavior for GetUserType is in fact holds.

Comment (SupinePandora43) [reply gm_dotnet_managed/Tests/GetUserTypeOnTypeMismatch.cs:1]: Oh ok

**Review by SupinePandora43 — APPROVED** [review]
Lgtm

**Review by AaronJThompson — APPROVED** [review]
Looks good. I Look forward to having docs 😅
